### ISSUE TITLE
fs:  uv_fs_read need do some housekeeper?

### ISSUE THREAD

**Issue by yjhjstz** [libuv/libuv#541]
```c
#define INIT(subtype)                                                         \
  do {                                                                        \
    req->type = UV_FS;                                                        \
    if (cb != NULL)                                                           \
      uv__req_init(loop, req, UV_FS);                                         \
    req->fs_type = UV_FS_ ## subtype;                                         \
    req->result = 0;                                                          \
    req->ptr = NULL;                                                          \
    req->loop = loop;                                                         \
    req->path = NULL;                                                         \
    req->new_path = NULL;                                                     \
    req->cb = cb;                                                             \
  }                                                                           \
  while (0)
```
did call `uv__req_init(loop, req, UV_FS); ` when cb isn't null.
```c
INIT(READ);
  req->file = file;

  req->nbufs = nbufs;
  req->bufs = req->bufsml;
  if (nbufs > ARRAY_SIZE(req->bufsml))
    req->bufs = uv__malloc(nbufs * sizeof(*bufs));

  if (req->bufs == NULL)
    return -ENOMEM;
```
But when `uv_fs_read` meat ENOMEM error , should we do some housekeeper that call 
`uv__req_unregister`?
Otherwise, `active_reqs` will explode.

**Comment by saghul** [comment]
Yes, we need to undo what `uv__req_init` did. Thanks for looking into these issues!

**Comment by yjhjstz** [comment]
yes, I will send PR,  but lots to fix.

**Comment by saghul** [comment]
:heart_eyes_cat:

**Comment by saghul** [comment]
Fixed by #543
